openrouteservice/src/main/java/heigit/ors/routing/algorithms/RPHASTAlgorithm.java

```diff
@@ -275,19 +290,18 @@ private void fillEdgesUpward(MultiTreeSPEntry currEdge, PriorityQueue<MultiTreeS
 						_msptItem = currEdge.getItem(i);
 						_entryWeight = _msptItem.weight;
 						
-						if (_entryWeight == 0.0)
+						if (_entryWeight == Double.POSITIVE_INFINITY)
```

Comment:
You should use Double.isInfinity instead

Reply:
If you look at the implementation of Double.isInfinite, it its
```public static boolean isInfinite(double var0) {
        return var0 == 1.0D / 0.0 || var0 == -1.0D / 0.0;
 } ```
which is 50% of the definition of Double.POSITIVE_INFINITY:
```public static final double POSITIVE_INFINITY = 1.0D / 0.0;```
As I do not want to accept Double.NEGATIVE_INFINITY, which would hint at an error in my code, I am happy to use my version, which should be faster as well.